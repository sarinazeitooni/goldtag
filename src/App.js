import Header from './components/header/Header';
import MainComponents from './components/MainComponents';
import Advices from './components/advices/advices';
import Footer from './components/footer/footer';
import Detail from './components/detailpage/allDetails/detail';
import './style/main.scss';
import {BrowserRouter,Route} from "react-router-dom";
import { Provider } from 'react-redux';
import store, {persistor} from './redux/store/store';

import { PersistGate } from 'redux-persist/integration/react';
function App() {
  return (
      <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
        <div className='app'>
          <Header/>
          <BrowserRouter>
                <Route exact path='/' component={MainComponents}/>
                <Route path='/:id' component={Detail}/>
          </BrowserRouter>
          <Advices/>
          <Footer/>
        </div>
          </PersistGate>
      </Provider>
  );
}
export default App;
