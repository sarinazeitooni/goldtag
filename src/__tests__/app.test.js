import React from 'react';
import {shallow} from 'enzyme';
import App from "../App";
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

describe("App component", () => {
    const store = configureStore();
    const initialState = {}
    const mockStore = configureStore();
    let Store = mockStore(initialState);
    it("it should render", () => {
        const container = shallow(<Provider store={Store}><App/></Provider>)
        expect(container).toMatchSnapshot();
    });
});