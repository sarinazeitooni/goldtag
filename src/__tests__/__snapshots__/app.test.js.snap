// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component it should render 1`] = `ShallowWrapper {}`;
